website2/src/components/documentationPage/documentHeader/index.tsx

```diff
@@ -1,30 +1,33 @@
 import { Text } from "@/components/text";
-import { PropsWithChildren, ReactNode } from "react";
+import { ReactNode } from "react";
 import styles from "./styles.module.css";
 import { Share } from "@/components/icons/share";
 import Link from "next/link";
+import { z } from "zod";
+import { checkPropsRuntime } from "@/utils/checkPropsRuntime";
 
-type Props = PropsWithChildren & {
-  type: "h1" | "h2" | "h3";
-  className?: string;
-};
+const PropSchema = z
+  .object({
+    type: z.enum(["h1", "h2", "h3"]),
+    className: z.string().optional(),
+    children: z.custom(),
+  })
+  .strict();
```

Comment:
@alexdavid i would like to call `.strict()` inside `checkPropsRuntime` but I can't get the typing to work out